rust/src/database/verkle/compute_commitment.rs

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2025 Sonic Operations Ltd
+//
+// Use of this software is governed by the Business Source License included
+// in the LICENSE file and at soniclabs.com/bsl11.
+//
+// Change Date: 2028-4-16
+//
+// On the date above, in accordance with the Business Source License, use of
+// this software will be governed by the GNU Lesser General Public License v3.
+
+use crate::{
+    database::verkle::crypto::{Commitment, Scalar},
+    types::Value,
+};
+
+/// Computes the commitment of a leaf node.
+///
+/// Since [`crate::database::verkle::crypto::Scalar`] cannot safely represent 32 bytes,
+/// the 256 32-bit values are split into two interleaved sets of 16 byte values, on which
+/// commitments C1 and C2 are computed separately:
+///
+///   C1 = Commit([  v[0][..16]),   v[0][16..]),   v[1][..16]),   v[1][16..]), ...])
+///   C2 = Commit([v[128][..16]), v[128][16..]), v[129][..16]), v[129][16..]), ...])
+///
+/// The final commitment of a leaf node is then computed as follows:
+///
+///    C = Commit([1, stem, C1, C2])
+///
+/// For details on the commitment procedure, see
+/// https://blog.ethereum.org/2021/12/02/verkle-tree-structure#commitment-to-the-values-leaf-nodes
+#[cfg_attr(not(test), expect(unused))]
+pub fn compute_leaf_node_commitment(
```

Comment:
This function needs tests.

Reply:
I added a test comparing against a hash generated with the Go reference implementation.